WikiWatershed/model-my-watershed
# Unit Conversion: MapShed

## Overview

Implements Unit Conversion for MapShed.

Connects #3024 

### Demo

![image](https://user-images.githubusercontent.com/1430060/49051146-94f08c80-f1b4-11e8-9171-2aea18581e02.png)

![image](https://user-images.githubusercontent.com/1430060/49051168-a9348980-f1b4-11e8-8314-99e7b5a4738b.png)

![image](https://user-images.githubusercontent.com/1430060/49051184-c10c0d80-f1b4-11e8-9022-c0de779374b9.png)

![image](https://user-images.githubusercontent.com/1430060/49051194-ce28fc80-f1b4-11e8-8396-48110565ceb9.png)

![image](https://user-images.githubusercontent.com/1430060/49051204-d97c2800-f1b4-11e8-91f6-88a5dc9d244c.png)

![image](https://user-images.githubusercontent.com/1430060/49051212-e4cf5380-f1b4-11e8-8434-f9dec134958b.png)

![image](https://user-images.githubusercontent.com/1430060/49051228-f57fc980-f1b4-11e8-96eb-db5b05b79a98.png)

![image](https://user-images.githubusercontent.com/1430060/49051256-092b3000-f1b5-11e8-8b82-d636b38fe52c.png)

![image](https://user-images.githubusercontent.com/1430060/49051269-1e07c380-f1b5-11e8-8546-c4b81c6cc5a8.png)

![image](https://user-images.githubusercontent.com/1430060/49051286-32e45700-f1b5-11e8-8f2b-c543efbb630a.png)

### Notes

They had asked to use US short ton for Sediments, since its values are so much larger than Nitrogen and Phosphorus which are better represented in pounds. However, some of our tables (particularly the summary table in Water Quality result tab in MapShed) put the units in the first column for each row, which could not resolve different cells having different units, so I've kept them all as pounds for now. This can be addressed at a later stage.

## Testing Instructions

* Checkout this branch and `bundle --debug`
* Login and set your units as "US Customary"
* Make a MapShed project for a well known area. On Staging, also make a MapShed project using the same well known area.
* As you go through MapShed, ensure that US Customary units are used throughout: in the results, in the compare tab, in manual entry, in land cover entry, in conservation practices modifications.
* Compare with the project on staging and ensure the conversions make sense
* Go to settings and change your units to Metric. Re-open the project. Ensure all results and your modifications are appropriately converted.
